### DISCUSSION TITLE
No tests match the provided pattern: ??

### DISCUSSION THREAD

**Discussion by ArlisVilipold** [Cyfrin/security-and-auditing-full-course-s23#236] [Q&A]
I have an issue running tests.. 
![test1](https://github.com/user-attachments/assets/650c1ab4-2921-4de6-b176-f28e2fd5eafe)

I'm 100% in right directory. 
![what](https://github.com/user-attachments/assets/d31caf4a-b73a-4274-a069-8d5e08892850)

no idea why is this happening.

`
function test_reentrancyRefund() public {

        address[] memory players = new address[](4);
        players[0] = playerOne;
        players[1] = playerTwo;
        players[2] = playerThree;
        players[3] = playerFour;
        puppyRaffle.enterRaffle{value: entranceFee * 4}(players);

        ReentrancyAttacker attackerContract = new ReentrancyAttacker(puppyRaffle);
        address attackUser = makeAddr("attackUser");
        vm.deal(attackUser, 1 ether);

        uint256 startingAttackerContractBalance = address(attackerContract).balance;
        uint256 startingPuppyRaffleBalance = address(puppyRaffle).balance;
        
        // attack 
        vm.prank(attackUser);
        attackerContract.attack{value: entranceFee}();

        console.log("here is the attacker contract balance", startingAttackerContractBalance);
        console.log("here is the puppy raffle balance", startingPuppyRaffleBalance);

        console.log("ending balance", startingPuppyRaffleBalance);
        console.log("ending balance of attacker contrat", startingAttackerContractBalance);

    }

`

**Comment by EngrPips** [comment]
Hello @ArlisVilipold, please run `foundryup`, then `forge clean` then retry.

**Reply by ArlisVilipold** [reply]
it does not work.

**Reply by EngrPips** [reply]
Does that happen for every single test or for that specific test?

**Reply by ArlisVilipold** [reply]
the 18 tests provided with gitclone works, this specific test does not work.

**Reply by EngrPips** [reply]
I haven't run into this issue before. It is hard for me to tell what is wrong.
